docs/block-development/extensible-blocks/cart-and-checkout-blocks/introduction-checkout-blocks/01-understanding-checkout-blocks.md

```diff
@@ -0,0 +1,430 @@
+# Understanding WooCommerce Checkout Blocks
+
+**Reading time: 5 minutes**
+
+> **💡 Compatibility Note:** This guide covers WooCommerce 9.0+ with Checkout Blocks (bundled since WooCommerce 8.3+). If you're using an older version, some features may not be available. Check your WooCommerce version: **WooCommerce → Settings → Status**.
+
+## What This Document Covers
+
+By the end of this guide, you'll understand:
+- The fundamental architectural shift from shortcode to blocks
+- Why WooCommerce made this change
+- The three core concepts powering checkout blocks
+- Whether checkout blocks are right for your project
+
+## The Shift: From Templates to Components
+
+### Shortcode Checkout (Traditional)
+
+The traditional WooCommerce checkout (`[woocommerce_checkout]`) works like a traditional web form:
+
+```
+Customer fills form → Clicks submit → Server validates → Creates order → Redirects
+```
+
+**Key characteristics:**
+- Server-rendered PHP templates
+- Sequential, synchronous processing
+- Validation happens on submission
+- Extensions modify output via template hooks and filters
+- State exists only during form submission
+
+### Checkout Blocks (Modern)
+
+The checkout block is a React-based application that maintains continuous state:
+
+```
+Customer interacts → State updates → Server syncs → UI updates → (repeat) → Order created
+```
+
+**Key characteristics:**
+- React components with real-time state management
+- Continuous synchronization with server
+- Validation happens as user types
+- Extensions use blocks, slots, and data stores
+- State persists throughout the checkout session
+
+## Why This Change?
```

Comment:
I'd remove this section. Overall, the docs should be actionable rather than theoretical/philosophical as much as possible.